src/diffusers/pipeline_utils.py

```diff
@@ -112,6 +113,31 @@ def save_pretrained(self, save_directory: Union[str, os.PathLike]):
             save_method = getattr(sub_model, save_method_name)
             save_method(os.path.join(save_directory, pipeline_component_name))
 
+
+    def to(self, torch_device: Optional[Union[str, torch.device]] = None):
+        modules, _ = self.extract_init_dict(dict(self.config))
+        for name in modules.keys():
+            module = getattr(self, name)
+            if isinstance(module, torch.nn.Module):
+                if torch_device is None:
+                    if module.device.type == "cpu":
+                        torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+                    else:
+                        torch_device = module.device
```

Comment:
I'm actually in favor of not doing anything when `torch_device`. Think it's better to have same semantics as the pt `.to`.